@dmarteau @elpaso (and potentially other server people), currently there is a build warning

```
In file included from ../src/server/qgsfilterrestorer.cpp:18:
../src/server/qgsfilterrestorer.h:75:29: warning: private field 'mAccessControl' is not used [-Wunused-private-field]
    const QgsAccessControl *mAccessControl = nullptr;
```
https://travis-ci.org/qgis/QGIS/jobs/324486415#L696-L698

which led me to think it's some leftover but now I realize that there's always this parameter specified in the constructor (see new warnings [in the last travis job](https://travis-ci.org/qgis/QGIS/builds/324620561?utm_source=github_status&utm_medium=notification)). Did I miss something obvious or can this be killed altogether?

--- a/src/server/qgsfilterrestorer.h
+++ b/src/server/qgsfilterrestorer.h
@@ -35,9 +35,7 @@ class SERVER_EXPORT QgsOWSServerFilterRestorer
 {
   public:
 
-    QgsOWSServerFilterRestorer( const QgsAccessControl *accessControl )
-      : mAccessControl( accessControl )
-    {}
+    QgsOWSServerFilterRestorer() = default;
 
     //! Destructor. When object is destroyed all original layer filters will be restored.
     ~QgsOWSServerFilterRestorer()
@@ -72,7 +70,6 @@ class SERVER_EXPORT QgsOWSServerFilterRestorer
     static void applyAccessControlLayerFilters( const QgsAccessControl *accessControl, QgsMapLayer *mapLayer );
 
   private:
-    const QgsAccessControl *mAccessControl = nullptr;
     QHash<QgsMapLayer *, QString> mOriginalLayerFilters;
 
 };

--- a/src/server/services/wfs/qgswfsgetfeature.cpp
+++ b/src/server/services/wfs/qgswfsgetfeature.cpp
@@ -181,7 +181,7 @@ namespace QgsWfs
 
     //scoped pointer to restore all original layer filters (subsetStrings) when pointer goes out of scope
     //there's LOTS of potential exit paths here, so we avoid having to restore the filters manually
-    std::unique_ptr< QgsOWSServerFilterRestorer > filterRestorer( new QgsOWSServerFilterRestorer( accessControl ) );
+    std::unique_ptr< QgsOWSServerFilterRestorer > filterRestorer( new QgsOWSServerFilterRestorer() );
 
     // features counters
     long sentFeatures = 0;

--- a/src/server/services/wfs/qgswfstransaction.cpp
+++ b/src/server/services/wfs/qgswfstransaction.cpp
@@ -239,7 +239,7 @@ namespace QgsWfs
 
     //scoped pointer to restore all original layer filters (subsetStrings) when pointer goes out of scope
     //there's LOTS of potential exit paths here, so we avoid having to restore the filters manually
-    std::unique_ptr< QgsOWSServerFilterRestorer > filterRestorer( new QgsOWSServerFilterRestorer( accessControl ) );
+    std::unique_ptr< QgsOWSServerFilterRestorer > filterRestorer( new QgsOWSServerFilterRestorer() );
 
     // get layers
     QStringList wfsLayerIds = QgsServerProjectUtils::wfsLayerIds( *project );

--- a/src/server/services/wfs/qgswfstransaction_1_0_0.cpp
+++ b/src/server/services/wfs/qgswfstransaction_1_0_0.cpp
@@ -226,7 +226,7 @@ namespace QgsWfs
 
       //scoped pointer to restore all original layer filters (subsetStrings) when pointer goes out of scope
       //there's LOTS of potential exit paths here, so we avoid having to restore the filters manually
-      std::unique_ptr< QgsOWSServerFilterRestorer > filterRestorer( new QgsOWSServerFilterRestorer( accessControl ) );
+      std::unique_ptr< QgsOWSServerFilterRestorer > filterRestorer( new QgsOWSServerFilterRestorer() );
 
       // get layers
       QStringList wfsLayerIds = QgsServerProjectUtils::wfsLayerIds( *project );

--- a/src/server/services/wms/qgswmsrenderer.cpp
+++ b/src/server/services/wms/qgswmsrenderer.cpp
@@ -123,7 +123,9 @@ namespace QgsWms
                             const QgsProject *project,
                             const QgsServerRequest::Parameters &parameters )
     : mParameters( parameters )
+#ifdef HAVE_SERVER_PYTHON_PLUGINS
     , mAccessControl( serverIface->accessControls() )
+#endif
     , mSettings( *serverIface->serverSettings() )
     , mProject( project )
   {

--- a/src/server/services/wms/qgswmsrenderer.h
+++ b/src/server/services/wms/qgswmsrenderer.h
@@ -286,8 +286,10 @@ namespace QgsWms
 
       const QgsServerRequest::Parameters &mParameters;
 
+#ifdef HAVE_SERVER_PYTHON_PLUGINS
       //! The access control helper
       QgsAccessControl *mAccessControl = nullptr;
+#endif
       QgsFeatureFilter mFeatureFilter;
 
       const QgsServerSettings &mSettings;

